File: SystemAPI/Modal/Endereco.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace SystemAPI.Modal
{
    [Table("dbo.Endereco")]
    public class Endereco
    {
        [Key]
        public string Id_Endereco { get; set; }
        
        public string Rua { get; set; }
        
        public string Complemento { get; set; }
        
        public string CEP { get; set; }
        
        public string Estado { get; set; }

        public string Cidade { get; set; }

        public Endereco( string idendereco, string rua, string cep, string estado, string cidade, string complemento) 
        {
            this.Id_Endereco = idendereco;
            this.Estado = estado;
            this.Rua = rua;
            this.CEP = cep;
            this.Cidade = cidade;
            this.Complemento = complemento;
        }

        public Endereco()
        {
        }
    }
}

File: Front/Cliente/ControlSystem/src/app/module/endereco/modal/endereco.ts
export class Endereco {
    readonly cep;
    readonly rua;
    readonly id_Endereco;
    readonly complemento;
    readonly cidade;
    readonly estado;

    
    public get CEP() : string {
        return this.cep;
    }
    
    
    public get Estado() : string {
        return this.estado;
    }
    
    
    public get Cidade() : string {
        return this.cidade;
    }
    
    public get IdEndereco() : string {
        return this.id_Endereco
    }

    public get Rua() : string {
        return this.rua;
    }
    
    
    public get Complemento() : string {
        return this.complemento;
    }
    
    
    

    constructor({ idEndereco, cep, rua, complemento, cidade, estado }:
        { idEndereco: string, cep: string, rua: string, complemento: string, 
        cidade: string, estado: string }) {
        this.cep = cep;
        this.cidade = cidade;
        this.complemento = complemento;
        this.estado = estado;
        this.id_Endereco = idEndereco;
        this.rua = rua;
    }
}